service/worker/replicator/replicator.go

```diff
@@ -55,21 +63,30 @@ type (
 	}
 )
 
+const (
+	replicationTimeout = 30 * time.Second
+)
+
 // NewReplicator creates a new replicator for processing replication tasks
 func NewReplicator(clusterMetadata cluster.Metadata, metadataManagerV2 persistence.MetadataManager,
-	historyClient history.Client, config *Config, client messaging.Client, logger bark.Logger,
-	metricsClient metrics.Client) *Replicator {
+	domainCache cache.DomainCache, clientBean client.Bean, historyClient history.Client,
```

Comment:
looks like client.Bean interface already supports creating all clients including client for remote cadence cluster.  Can we just use this to initialize other clients like historyClient instead of explicitly passing in.